musicbrainz/csv/fetch.py

```diff
@@ -0,0 +1,59 @@
+"""
+fetch the latest test data dump files
+"""
+
+import os
+import time
+import requests
+from bs4 import BeautifulSoup
+
+
+def get_latest_json_dump_url():
+    """
+    get the latest repo
+    """
+    url = "https://data.metabrainz.org/pub/musicbrainz/data/json-dumps/"
+    response = requests.get(url, timeout=50)
+    soup = BeautifulSoup(response.text, "html.parser")
+
+    # Extract all directories
+    dirs = [a.text for a in soup.find_all("a") if a.text.startswith("latest-is-")]
+
+    latest_dir = dirs[-1]
+    latest_dir = latest_dir[10:]
+    resp_url = url + latest_dir + "/"
+    return resp_url
+
+
+def fetch_api_call(url):
+    """
+    API calls to fetch all .tar.xz files
+    """
+    response = requests.get(url, timeout=50)
+    soup = BeautifulSoup(response.content, "html.parser")
+
+    # Find all links ending with .tar.xz
+    tar_xz_files = [
+        url + a["href"] for a in soup.find_all("a") if a["href"].endswith(".tar.xz")
+    ]
```

Comment:
I think you should probably get these files by name, rather than grab all of them.

Reply:
I don't understand what you mean by getting them by name. Are you saying getting them by names like "recording.tar.xz"?

Reply:
Yes. Like, have a list of names of the files you are trying to download and run through that list.

Reply:
Ok. I will change this.